src/types/UiaTextRangeBase.cpp

```diff
@@ -1009,13 +1009,21 @@ std::wstring UiaTextRangeBase::_getTextValue(til::CoordType maxLength) const
         textData.reserve(textDataSize);
         for (const auto& text : bufferData.text)
         {
+            if (textData.size() >= maxLength)
+            {
+                // early exit; we're already at/past max length
+                break;
+            }
             textData += text;
         }
     }
 
-    if (maxLength >= 0)
+    // - size_t(-1) gets converted to 0xffff...
+    // - if the maxLength is more than what we have,
+    //     they probably just want the whole thing (so don't resize)
+    if (const auto s = gsl::narrow_cast<size_t>(maxLength); s < textData.size())

```

Comment:
I legitimately think that you could (should?) rework the text append loop to not ever append more text than the user requests. After all, why should we copy in a bunch of data that we're just going to delete?

(e.g. If the width is 120, and maxLength is 121... you're going to loop twice:

```
Append 120.
Append 120.
Exit loop.
Truncate from 240 to 121.
```

Why copy the extra 119 characters?
)

Reply:
If you never append more than they asked for, you never have to shrink it.

Reply:
We have to add the `\r\n` at the end of each line. We could probably do the math early and account for that, but I figured a resize wouldn't hurt knowing that we'll be downsizing (instead of going up and adding all the `L'\0'`)